src/Components/Components/src/Rendering/Renderer.cs

```diff
@@ -42,6 +57,22 @@ public Renderer(IServiceProvider serviceProvider)
             _componentFactory = new ComponentFactory(serviceProvider);
         }
 
+        /// <summary>
+        /// Constructs an instance of <see cref="Renderer"/>.
+        /// </summary>
+        /// <param name="serviceProvider">The <see cref="IServiceProvider"/> to be used when initialising components.</param>
+        public Renderer(IServiceProvider serviceProvider, IDispatcher dispatcher)
+        {
+            _componentFactory = new ComponentFactory(serviceProvider);
```

Comment:
Could we chain this constructor to the single-parameter one to avoid repeating this line? I know it's not a lot of code to repeat, but I think it would make the intent clearer.

i.e.,

```
public Renderer(IServiceProvider serviceProvider, IDispatcher dispatcher) : this(serviceProvider)
{
    _dispatcher = dispatcher;
}
```

Reply:
I didn't chain it so that the linker could strip this one completely away. I can do so for now, but I think we'll end up changing it back.

Reply:
I don't follow. Wouldn't the linker want to remove the 2-param one in the WebAssembly case? The fact that it calls the 1-param one wouldn't stop the linker from removing it. Am I misunderstanding what gets called?

Reply:
i read this wrong, as discussed offline, I will change it